uk.co.compucorp.civicrm.hrcore/gulp/tasks/backstopjs.js

```diff
@@ -174,37 +170,31 @@ function runBackstopJS (command) {
   if (touchSiteConfigFile()) {
     utils.throwError(
       'No site-config.json file detected!\n' +
-      '\tOne has been created for you under ' + path.basename(BACKSTOP_DIR) + '/\n' +
+      `\tOne has been created for you under ${path.basename(BACKSTOP_DIR)}/\n` +
       '\tPlease insert the real value for each placeholder and try again'
     );
   }
 
-  return writeCookies()
-    .then(createTempConfigFile)
-    .then(function (tempConfigFile) {
-      return new Promise(function (resolve, reject) {
-        var success = false;
-
-        gulp.src(FILES.tpl)
-          .pipe(file(path.basename(FILES.temp), tempConfigFile))
-          .pipe(gulp.dest(BACKSTOP_DIR))
-          .on('end', function () {
-            backstopjs(command, {
-              configPath: FILES.temp,
-              filter: argv.filter
-            })
-              .then(function () {
-                success = true;
-              })
-              .catch(_.noop).then(function () { // equivalent to .finally()
-                cleanUpAndNotify(success);
-
-                success ? resolve() : reject(new Error('BackstopJS error'));
-              });
-          });
+  return new Promise((resolve, reject) => {
+    let success = false;
+
+    gulp.src(FILES.tpl)
+      .pipe(file(path.basename(FILES.temp), createTempConfig()))
+      .pipe(gulp.dest(BACKSTOP_DIR))
+      .on('end', async () => {
+        try {
+          (typeof argv.skipCookies === 'undefined') && await writeCookies();
+          await backstopjs(command, { configPath: FILES.temp, filter: argv.filter });
+
+          success = true;
+        } finally {
+          cleanUpAndNotify(success);
+
+          success ? resolve() : reject(new Error('BackstopJS error'));
```

Comment:
Could we provide a more specific error message?

Reply:
unfortunately not, it needs to be generic as the error could be

1. Task error
2. Script errors
3. Comparison errors

Reply:
Please include this info in the error message.